fury/actors/peak.py

```diff
@@ -0,0 +1,372 @@
+
+        vs_dec_code = load('peak_dec.vert')
+        vs_impl_code = load('peak_impl.vert')
+        fs_dec_code = load('peak_dec.frag')
+        fs_impl_code = load('peak_impl.frag')
+
+        shader_to_actor(self, 'vertex', decl_code=vs_dec_code,
+                        impl_code=vs_impl_code)
+        shader_to_actor(self, 'fragment', decl_code=fs_dec_code)
+        shader_to_actor(self, 'fragment', impl_code=fs_impl_code,
+                        block='light')
+
+        # Color scale with a lookup table
+        if colors_are_scalars:
+            if lookup_colormap is None:
+                lookup_colormap = colormap_lookup_table()
+
+            self.__mapper.SetLookupTable(lookup_colormap)
+            self.__mapper.UseLookupTableScalarRangeOn()
+            self.__mapper.Update()
+
+        self.__lw = linewidth
+        self.GetProperty().SetLineWidth(self.__lw)
+
+        if self.__global_opacity >= 0:
+            self.GetProperty().SetOpacity(self.__global_opacity)
+
+        self.__min_centers = np.min(indices, axis=1)
+        self.__max_centers = np.max(indices, axis=1)
+
+        self.__is_range = True
+        self.__low_ranges = self.__min_centers
+        self.__high_ranges = self.__max_centers
+        self.__cross_section = self.__high_ranges // 2
+
+        self.__mapper.AddObserver(vtk.vtkCommand.UpdateShaderEvent,
+                                  self.__display_peaks_vtk_callback)
+
+    @vtk.calldata_type(vtk.VTK_OBJECT)
+    def __display_peaks_vtk_callback(self, caller, event, calldata=None):
+        if calldata is not None:
+            calldata.SetUniformi('isRange', self.__is_range)
+            calldata.SetUniform3f('highRanges', self.__high_ranges)
+            calldata.SetUniform3f('lowRanges', self.__low_ranges)
+            calldata.SetUniform3f('crossSection', self.__cross_section)
+
+    def display_cross_section(self, x, y, z):
+        if self.__is_range:
+            self.__is_range = False
+        self.__cross_section = [x, y, z]
+
+    def display_extent(self, x1, x2, y1, y2, z1, z2):
+        if not self.__is_range:
+            self.__is_range = True
+        self.__low_ranges = [x1, y1, z1]
+        self.__high_ranges = [x2, y2, z2]
+
+    @property
+    def cross_section(self):
+        return self.__cross_section
+
+    @property
+    def global_opacity(self):
+        return self.__global_opacity
+
+    @global_opacity.setter
+    def global_opacity(self, opacity):
+        self.__global_opacity = opacity
+        self.GetProperty().SetOpacity(self.__global_opacity)
+
+    @property
+    def high_ranges(self):
+        return self.__high_ranges
+
+    @property
+    def is_range(self):
+        return self.__is_range
+
+    @is_range.setter
+    def is_range(self, is_range):
+        self.__is_range = is_range
+
+    @property
+    def low_ranges(self):
+        return self.__low_ranges
+
+    @property
+    def linewidth(self):
+        return self.__lw
+
```

Comment:
Can you cover these property tests too?

Reply:
Done, I have reviewed all the code coverage and addressed all the issues.